packages/core/src/services/essentials/dom-utils.ts

```diff
@@ -46,34 +63,39 @@ export class DOMUtils extends Service {
 			height: svgSelector.node().clientHeight
 		};
 
-		const validateDimensions = dimensions => {
-			if (dimensions && dimensions.width && dimensions.height &&
-				!isNaN(dimensions.height) && !isNaN(dimensions.width) &&
-				("" + dimensions.width + dimensions.height).indexOf("%") === -1 &&
-				dimensions.width > 0 && dimensions.height > 0) {
-				return true;
-			}
-
-			return false;
-		};
-
 		// If both attribute values are numbers
 		// And not percentages or NaN
 		if (options) {
-			if (options.useAttrs && validateDimensions(attrDimensions)) {
-				return attrDimensions;
+			if (options.useAttrs) {
+				validateAndSetDimensions(attrDimensions);
+
+				if (finalDimensions.width > 0 && finalDimensions.height > 0) {
```

Comment:
Can these checks be part of `validateAndSetDimensions` so you can just `return validateAndSetDimensions(...)`?
It doesn't seem like it would affect the cases that don't use a return value.

Reply:
This is really about priority of alternative calculations within the function.

If I always `return validateAndSet...()` then none of the sections below it would execute